### ISSUE TITLE
aws_ecs: allow to specify kernel when creating the ecsOptimizedImage

### ISSUE THREAD

**Issue by segundovolante** [aws/aws-cdk#29800]
### Describe the feature

Current CDK api does not allow to specify the kernel version, and you end up with 4.14 which is marked as EOL Jan 2024.

This page says 4.14 has reached EOL: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-optimized_AMI.html 

API link: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ecs.EcsOptimizedImage.html 

It would help if:
*  default kernel is the recommended kernel
* it could be passed like in this construct: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.AmazonLinux2Kernel.html 
* create the AMI id using https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.AmazonLinux2ImageSsmParameter.html 

### Use Case

I want to create an ECS autoscaling group with the recommended kernel or using another supported ami from the ECS optimized image.

### Proposed Solution

Option 1: update the api to use the recommended kernel version (5.10)
Option 2: allow users to pass the kernel version to use, as the AMI exists. 

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.137.0

### Environment details (OS name and version, etc.)

AL2

**Comment by khushail** [comment]
Thanks @segundovolante for requesting this feature. This totally makes sense!

**Comment by Konosh93** [comment]
> it could be passed like in this construct: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.AmazonLinux2Kernel.html

I concur with the proposal to allow manual specification of the kernel version in ecsOptimizedImage. While updating the default kernel version to 5.10 might seem beneficial, it does carry the risk of introducing breaking changes for current users.

Additionally, changing the default kernel would necessitate updates to a large number of integration tests

**Comment by segundovolante** [comment]
Yeah, I agree that the better option long term is to allow to specify the kernel as an optional parameter. This would behave similar to the ec2.
